library/Vanilla/Embeds/ImgurEmbed.php

```diff
@@ -0,0 +1,70 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0
+ */
+
+namespace Vanilla\Embeds;
+
+/**
+ * Instagram Embed.
+ */
+class ImgurEmbed extends Embed {
+
+    /** @inheritdoc */
+    protected $domains = ['imgur.com'];
+
+    /**
+     * ImgurEmbed constructor.
+     */
+    public function __construct() {
+        parent::__construct('imgur', 'image');
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function matchUrl(string $url) {
+        $data = [];
+
+        if ($this->isNetworkEnabled()) {
+            preg_match('/https?:\/\/imgur\.com\/gallery\/(?<postID>[a-z0-9]+)/i', $url, $matches);
```

Comment:
This doesn't seem to be as flexible as it should be. It works on "gallery" URLs, but not other types of URLs. For example:

1. Works: https://imgur.com/gallery/10HROiq
1. Does not work: https://imgur.com/10HROiq
1. Does not work: https://m.imgur.com/gallery/10HROiq
1. Does not work: https://imgur.com/a/LQVun5v